### ISSUE TITLE
`nginx.<vhost>.enableSSL` also disables non-SSL HTTP connections, breaks ACME

### ISSUE THREAD

**Issue by michaelpj** [NixOS/nixpkgs#27338]
## Issue description
I expected adding `enableSSL` to just enable SSL connections, but it actually replaces the normal connection entirely.

This causes problems in conjunction with `enableACME` - the request for the challenge response comes in over normal HTTP, so it gets rejected. Adding `forceSSL` makes it work, because the request is then redirected to HTTPS which actually works.

I think a sensible fix would be for `enableSSL` to simply *add* the additional listen port, and for `forceSSL` to remove it in conjunction with adding the redirect.

### Steps to reproduce

```
services.nginx = {
  enable = true;
  virtualHosts.whatever = {
    enableSSL = true;
    enableACME = true;
  };
};
```

## Technical details

* System: 17.09pre110213.01c3847b9c
* Nix version: 1.11.11
* Nixpkgs version: 17.09pre110213.01c3847b9c
* Sandboxing enabled: true

**Comment by fadenb** [comment]
Related issues: https://github.com/NixOS/nixpkgs/issues/23711 https://github.com/NixOS/nixpkgs/issues/25533 https://github.com/NixOS/nixpkgs/pull/25604

**Comment by rnhmjoj** [comment]
Thanks! I will implement @globin three-option suggestion tomorrow. It's easy now with `vhost.listen`.

**Comment by wmertens** [comment]
fixed with #27426

**Comment by michaelpj** [comment]
Thanks!
